python/packages/ai/teams/ai/augmentations/tools_augmentation.py

```diff
@@ -0,0 +1,198 @@
+                                valid_action_tool_calls.append(curr_tool_call)
+                        else:
+                            valid_action_tool_calls.append(curr_tool_call)
+                    else:
+                        return Validation(valid=False, feedback="The invoked tool does not exist.")
+
+                # Calling multiple tools
+                else:
+                    for tool_call in tool_calls:
+                        function_name = tool_call.function.name
+                        curr_tool = None
+
+                        for tool in tools:
+                            if tool.name == function_name:
+                                curr_tool = tool
+                                break
+
+                        # Validate function name
+                        if not curr_tool:
+                            continue
+
+                        # Validate function arguments
+                        required_args = (
+                            curr_tool.parameters["required"]
+                            if curr_tool.parameters and "required" in curr_tool.parameters
+                            else None
+                        )
+
+                        curr_args = json.loads(tool_call.function.arguments)
+
+                        if required_args:
+                            # Contains all required args
+                            if all(args in curr_args for args in required_args):
+                                valid_action_tool_calls.append(tool_call)
+                        else:
+                            valid_action_tool_calls.append(tool_call)
+
+            # No tools were valid
+            if len(valid_action_tool_calls) == 0:
+                memory.set(SUBMIT_TOOL_OUTPUTS_VARIABLE, False)
+                memory.set(SUBMIT_TOOL_OUTPUTS_MAP, {})
+                memory.set(SUBMIT_TOOL_OUTPUTS_MESSAGES, [])
+                memory.set(SUBMIT_TOOL_HISTORY, [])
+
+            return Validation(
+                valid=True,
+                value=valid_action_tool_calls if len(valid_action_tool_calls) > 0 else None,
+            )
+        return Validation(valid=True)
+
+    async def create_plan_from_response(
+        self,
+        turn_context: TurnContext,
+        memory: MemoryBase,
+        response: PromptResponse[Union[str, List[chat.ChatCompletionMessageToolCall]]],
+    ) -> Plan:
+        """
+        Creates a plan given validated response value.
+
+        Args:
+            turn_context (TurnContext): Context for the current turn of conversation.
+            memory (MemoryBase): Interface for accessing state variables.
+            response (PromptResponse[Union[str, List[chat.ChatCompletionMessageToolCall]]]):
+                The validated and transformed response for the prompt.
+
+        Returns:
+            Plan: The created plan.
+        """
+
+        tool_map: Dict = {}
+        commands: List[Union[PredictedDoCommand, PredictedSayCommand]] = []
```

Comment:
Typescript has a type called `PredictedCommand` which is already a Union type -- you should be able to use that instead of defining a new Union type here.